### ISSUE TITLE
HackerOne TOTP is instead seen as password field

### ISSUE THREAD

**Issue by bwbroersma** [keepassxreboot/keepassxc-browser#2332]
The current (far from ideal) HackerOne TOTP:
```html
<input autocomplete="off" class="text-field__input" id="sign_in_totp_code" maxlength="6" name="user[totp_code]" type="password" value="">
```
Of course they should use `autocomplete="one-time-code"`. However the `/\btotp\b/` match in combination with `maxlength=6`, I think KeePassXC-browser should be enough hints to correctly detect TOTP here.

## Expected Behavior
Detect TOTP field.

## Current Behavior
The field is detected as password field.

## Possible Solution
```js
kpxcTOTPIcons.isAcceptedTOTPField(document.getElementById("sign_in_totp_code"))
```

is `true`, however because it's detected as a `password` field first, it seems the field is no longer detected as `TOTP`:
https://github.com/keepassxreboot/keepassxc-browser/blob/aa288ff797c0d9944734d7e08a6943b8715085d3/keepassxc-browser/content/fields.js#L23-L33

Maybe explicitly check for `!isAcceptedTOTPField` before adding it as a password?

## Steps to Reproduce (for bugs)
Enable 2FA on https://hackerone.com/ and sign in.

## Debug info
KeePassXC - 2.7.9
KeePassXC-Browser - 1.9.3
Operating system: Linux x86_64
Browser: Mozilla Firefox 131.0

**Comment by bwbroersma** [comment]
I've a PR for solution 1.

**Comment by droidmonkey** [comment]
That looks like a good solution to me

**Comment by bwbroersma** [comment]
I agree with you both @droidmonkey and @varjolintu that this should be seriously tested.
I so see TOTP having the `type=password` more often nowadays, e.g. also on namesilo.com 2FA:
```html
<label data-v-0a2585d0="" data-v-779d6dc5="" class=""><span data-v-0a2585d0="" data-v-779d6dc5="">2-Factor Code:</span><input data-v-0a2585d0="" data-v-779d6dc5="" type="password" maxlength="12" class="form-control" autocomplete="off"></label>
```

**Comment by droidmonkey** [comment]
Geez and that one had a max length 12 too. So weird.

**Comment by varjolintu** [comment]
> I agree with you both @droidmonkey and @varjolintu that this should be seriously tested. I do see TOTP having the `type=password` more often nowadays, e.g. also on namesilo.com 2FA:
> 
> ```
> <label data-v-0a2585d0="" data-v-779d6dc5="" class=""><span data-v-0a2585d0="" data-v-779d6dc5="">2-Factor Code:</span><input data-v-0a2585d0="" data-v-779d6dc5="" type="password" maxlength="12" class="form-control" autocomplete="off"></label>
> ```

Usually `type=password` fields are masked and useless for TOTP input.
